### DISCUSSION TITLE
Developer setup has problems

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [outline/outline#5681] [General]
- The local development doc is missing instructions on generating secrets and mapping `app.outline.dev` to localhost
- Got env validation errors about IFRAMELY_URL with no explanation on what this should be
- The docker-compose.yml exposes redis, postgres and s3 services on 127.0.0.1. This led to Database connection refused errors because my ISP uses IPv6 by default so localhost IP is `::1`

**Comment by apoorv-mishra** [comment]
Hey, sorry for the inconvenience. 

* What secrets are you referring to exactly? The `SECRET_KEY` inside `.env`? Also, the `app.outline.dev` changes were pushed not so long ago, agreed that the docs certainly need to be updated on that.
* It'd be good if you could point out what validation errors you got for `IFRAMELY_URL` and how. This too was pushed recently.
* Interesting! This needs to be looked into. Also, happy to know how you got this working, if at all you did.

**Comment by chengguizi** [comment]
I have the same issue regarding IFRAMELY_URL . The issue is basically that the string cannot be undefined, but we don't know what the value should be.

**Reply by tommoor** [reply]
The config has a default value, so this doesn't make any sense… 🤔

**Comment by mghlb** [comment]
The IFRAMELY_URL error gets resolved by removing it (if not needed) from .env, but I also have been trying to set a developer environment up for a couple of days now to no avail. Everything seems to compile just fine, but access points return nothing.
